### PR TITLE
Fix inconsistent dtype for reward in MemoryChain environment

### PR DESCRIPTION
**Pull request by smonsays**
Hey Robert,
I tried to `jax.lax.scan` over the bsuite-MemoryChain environment and noticed that the reward returned by this env is an `int32` instead of a `float32` as for the other environments. Was this on purpose? If not here would be the minimal fix.

Best,
Simon

### CODE REVIEW

**Comment by RobertTLange** [comment]
Thank you Simon!
